api/layer/object_cache.go

```diff
@@ -74,6 +86,29 @@ func (l *listObjectsCache) Put(key string, objects []*ObjectInfo) {
 	})
 }
 
-func createKey(accessKey string, cid *cid.ID) string {
-	return accessKey + cid.String()
+func createKey(ctx context.Context, cid *cid.ID) (string, error) {
+	box, err := GetBoxData(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return box.Gate.AccessKey + cid.String(), nil
+}
+
+func trimAfterObjectName(startAfter string, objects []*ObjectInfo) []*ObjectInfo {
+	for i := range objects {
+		if objects[i].Name > startAfter {
+			return objects[i:]
+		}
+	}
+	return nil
+}
+
+func trimAfterObjectID(id string, objects []*ObjectInfo) []*ObjectInfo {
+	for i, obj := range objects[:len(objects)-1] {
```

Comment:
Should we iterate over all objects? We do neofs `objectSearch` If continuation token is the last object id but I suppose we can return empty list to client.

Reply:
It  seems like a minor optimization, but added this.

Reply:
If we want to avoid redundant `search` we should change:
```
func trimAfterObjectID(id string, objects []*ObjectInfo) []*ObjectInfo {
	if len(objects) != 0 && objects[len(objects)-1].id.String() == id {
		return nil
	}
       ...
}
``` 
to 
```
func trimAfterObjectID(id string, objects []*ObjectInfo) []*ObjectInfo {
	if len(objects) != 0 && objects[len(objects)-1].id.String() == id {
		return []*ObjectInfo{}
	}
       ...
}
``` 
because there is check `if allObjects == nil ` in `ListObjectsV2`

Reply:
The idea was to try update data if we don't receive any from cache, but in case of ContinuationToken it's poinless maybe, fixed.

Reply:
Okay, fixed